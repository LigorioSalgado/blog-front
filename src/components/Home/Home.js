import React, { Component } from 'react';
import { API } from '../../utils/http';


const displayCards = (title) => {
	return (
		<div className="col-md-3">
			<div className="card" >
				<img className="card-img-top" src="https://picsum.photos/200/200" />
				<div className="card-body">
					<h5 className="card-title">{title}</h5>
				</div>
			</div>
		</div>
	)
}


class Home extends Component {

	constructor() {
		super();

		this.state = {
			posts: [],
			isLoading:true
		}
	}

	componentDidMount() {
		API.get('/posts').then((response) => {
			this.setState({isLoading:false, posts: response.data });
		}).catch((err) => {
			// eslint-disable-next-line no-console
			console.log(err);
		})

	}

	render() {
		const {posts,isLoading} =  this.state
		const cardPost = posts.map((post) => {
			return displayCards(post.title);
		})
		return (
			<div className="container">
				<h2>Post recientes</h2>
				<div className="row mt-4">
					{
						(isLoading) ? (<h3>Cargando...</h3>) : (cardPost)	
					}
				</div>
			</div>
		)
	}
}

export default Home;
