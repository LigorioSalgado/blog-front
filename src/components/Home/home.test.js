import React from 'react';
import { shallow, mount } from "enzyme";

import { Home } from '../Home';



describe("<Card/>", () => {

	it("Debe ejecutar render correctamente", () => {
		const component = shallow(<Home />)
		expect(component).toMatchSnapshot();
	})

	it("Debe crear cards", () => {

		const posts = [
			{ title: "prueba 1", image: "imagen prueba" },
			{ title: "prueba 2", image: "imagen prueba" },
			{ title: "prueba 3", image: "imagen prueba" }

		]
		const component = mount(<Home />);
		component.setState({ posts });
		expect(component.state().posts).toHaveLength(3)

	})

})
