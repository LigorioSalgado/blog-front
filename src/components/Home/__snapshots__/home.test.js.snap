// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Debe ejecutar render correctamente 1`] = `
<div
  className="container"
>
  <h1>
    Post recientes
  </h1>
  <div />
  <div
    className="row mt-4"
  >
    <h3>
      Cargando...
    </h3>
  </div>
</div>
`;
